diff --git a/platform/qemu/sifive_u/platform.c b/platform/qemu/sifive_u/platform.c
index 40d6584..769c3f5 100644
--- a/platform/qemu/sifive_u/platform.c
+++ b/platform/qemu/sifive_u/platform.c
@@ -18,7 +18,7 @@
 /* clang-format off */
 
 #define SIFIVE_U_HART_COUNT			4
-#define SIFIVE_U_HART_STACK_SIZE		8192
+#define SIFIVE_U_HART_STACK_SIZE		16384
 
 #define SIFIVE_U_SYS_CLK			1000000000
 #define SIFIVE_U_PERIPH_CLK			(SIFIVE_U_SYS_CLK / 2)
diff --git a/platform/qemu/virt/platform.c b/platform/qemu/virt/platform.c
index 6ef6a19..fe5a8a7 100644
--- a/platform/qemu/virt/platform.c
+++ b/platform/qemu/virt/platform.c
@@ -20,7 +20,7 @@
 /* clang-format off */
 
 #define VIRT_HART_COUNT			8
-#define VIRT_HART_STACK_SIZE		8192
+#define VIRT_HART_STACK_SIZE		16384
 
 #define VIRT_TEST_ADDR			0x100000
 #define VIRT_TEST_FINISHER_FAIL		0x3333
diff --git a/platform/sifive/fu540/platform.c b/platform/sifive/fu540/platform.c
index 81e8e8d..929fb70 100644
--- a/platform/sifive/fu540/platform.c
+++ b/platform/sifive/fu540/platform.c
@@ -22,7 +22,7 @@
 /* clang-format off */
 
 #define FU540_HART_COUNT			5
-#define FU540_HART_STACK_SIZE			8192
+#define FU540_HART_STACK_SIZE			16384
 
 #define FU540_SYS_CLK				1000000000
 
